src/components/subscribe/subscribe-form.jsx

```diff
@@ -0,0 +1,88 @@
+import React, { useState } from "react";
+import PropTypes from "prop-types";
+
+import { Hero } from "../hero/hero";
+import { Input } from "../input/input";
+import { Button } from "../button/button";
+
+import "./subscribe-form.scss";
+
+import { validateEmail } from "./email-validator.service";
+import { Wording } from "./constants";
+
+export const SubscribeForm = ({
+  compact,
+  sending,
+  success,
+  errorMessage,
+  onSubmitted
+}) => {
+  const [inputValue, setInputValue] = useState();
+  const [isValid, setIsValid] = useState();
+
+  const handleInputChange = ({ target }) => {
+    setIsValid(validateEmail(target.value));
```

Comment:
Showing the validation while the user is still typing is not a very good pattern. It's called "Don't scream at the user". We should show the validation only after the user click on the CTA button 🙌